packages/shield-controller/src/backend.ts

```diff
@@ -236,6 +266,30 @@ export class ShieldRemoteBackend implements ShieldBackend {
       Authorization: `Bearer ${accessToken}`,
     };
   }
+
+  #cancelPreviousPendingRequests(id: string) {
+    const abortController = this.#abortControllerMap.get(id);
+    if (abortController && !abortController.signal.aborted) {
+      abortController.abort();
+      this.#removeAbortHandler(id);
+    }
+  }
+
+  #removeAbortHandler(requestId: string, abortHandler?: () => void) {
+    const abortController = this.#abortControllerMap.get(requestId);
+    if (abortController) {
+      if (abortHandler) {
+        abortController.signal.removeEventListener('abort', abortHandler);
```

Comment:
I'm wondering: Is it necessary to remove the handler, if we drop the whole controller anyways?